R/tables.R

```diff
@@ -2,12 +2,13 @@
 MB = NCOL = NROW = NULL
 
 tables = function(mb=TRUE, order.col="NAME", width=80,
-                   env=parent.frame(), silent=FALSE, index=FALSE)
+                  env=parent.frame(), silent=FALSE, index=FALSE)
 {
   # Prints name, size and colnames of all data.tables in the calling environment by default
-  all_obj = objects(envir=env, all.names=TRUE)
-  is_DT = which(vapply_1b(all_obj, function(x) is.data.table(get(x, envir=env))))
-  if (!length(is_DT)) {
+  # include "hidden" objects (starting with .) via all.names=TRUE, but exclude ... specifically, #5197
+  all_obj = grep("...", objects(envir=env, all.names=TRUE, sorted=order.col == "NAME"), invert=TRUE, fixed=TRUE, value=TRUE)
```

Comment:
we could also pass `pattern="^([^.]|[.][^.]|[.][.][^.])"` to do it in one call, but that's pretty ugly, and anyway `objects()` itself is just calling `grep()` so there's no efficiency benefit (i.e., `pattern` is not being handled in C), so wrapping in `grep()` does the same thing with a more readable `grep()` call.

Reply:
`objects()` only calls `grep()` when `pattern` is not `missing`, iiuc. But `pattern` is missing in this `objects()` call. So doesn't adding `grep` here add overhead of grep?